mesonpy/__init__.py

```diff
@@ -427,48 +473,19 @@ def _install_path(
         Some files might need to be fixed up to set the RPATH to the internal
         library directory on Linux wheels for eg.
         """
+        builddir_to_installdir_mapping = self._project._copy_files
+        origin = pathlib.Path(builddir_to_installdir_mapping[os.fspath(path_in_builddir)])
         location = destination.as_posix()
         counter.update(location)
 
-        # fix file
-        if os.path.isdir(origin):
-            for root, dirnames, filenames in os.walk(str(origin)):
-                # Sort the directory names so that `os.walk` will walk them in a
-                # defined order on the next iteration.
-                dirnames.sort()
-                for name in sorted(filenames):
-                    path = os.path.normpath(os.path.join(root, name))
-                    if os.path.isfile(path):
-                        arcname = os.path.join(destination, os.path.relpath(path, origin).replace(os.path.sep, '/'))
-                        wheel_file.write(path, arcname)
+        if origin.is_dir():
+            for child in origin.rglob('**/*'):
+                if child.is_file():
+                    arcname = destination / child.relative_to(origin)
```

Comment:
Aren't the paths returned by `pathlib.Path.rglob()` already relative to `origin`?